src/main/java/model/CodeQualityRule.java

```diff
@@ -100,4 +103,9 @@ public String toString() {
 		return this.name;
 	}
 
+	public boolean equals(CodeQualityRule other) {

```

Comment:
conseguiste confirmar que isto está a funcionar como deve ser?

Reply:
Sim, now it is. Como fizemos com o cast da classe ficava a me lançar erros. I gave up and did this.